scripts/compare-memory-usage.ts

```diff
@@ -0,0 +1,18 @@
+const threshold = 90;
+
+const main = () => {
+  const v1MemoryUsage = parseInt(process.argv[2]);
+  const canaryMemoryUsage = parseInt(process.argv[3]);
+  const percentageBasedOnV1 = (canaryMemoryUsage / v1MemoryUsage) * 100;
+  if (percentageBasedOnV1 > threshold) {
+    throw new Error(
+      `Test failed: v1 memory usage is ${v1MemoryUsage} and canary memory usage is ${canaryMemoryUsage} (${percentageBasedOnV1} %). Consider change the threshold (${threshold} %) or fix it.`
+    );
+  }
+  /* eslint-disable-next-line no-console */
+  console.log(
+    `Test passed: v1 memory usage is ${v1MemoryUsage} and canary memory usage is ${canaryMemoryUsage} (${percentageBasedOnV1} %).`
```

Comment:
I think the percentage of the diff might be more interesting than the actual comparison (-10%, e.g.).

Reply:
So you prefer -10% rather than 90%?

Reply:
Discussed